controllers/ppwebhook.go

```diff
@@ -0,0 +1,470 @@
+		"openshift-kube-apiserver-operator",
+		"openshift-kube-controller-manager",
+		"openshift-kube-scheduler",
+		"openshift-kube-scheduler-operator",
+		"openshift-kube-storage-version-migrator",
+		"openshift-kube-storage-version-migrator-operator",
+		"openshift-machine-api",
+		"openshift-machine-config-operator",
+		"openshift-marketplace",
+		"openshift-monitoring",
+		"openshift-multus",
+		"openshift-network-diagnostics",
+		"openshift-network-operator",
+		"openshift-node",
+		"openshift-nutanix-infra",
+		"openshift-oauth-apiserver",
+		"openshift-openstack-infra",
+		"openshift-operator-lifecycle-manager",
+		"openshift-operators",
+		"openshift-ovirt-infra",
+		"openshift-ovn-kubernetes",
+		"openshift-route-controller-manager",
+		"openshift-service-ca",
+		"openshift-service-ca-operator",
+		"openshift-user-workload-monitoring",
+		"openshift-vsphere-infra",
+		"kube-system",
+		"kube-node-lease",
+	}
+
+	webhookSvcNamespace := os.Getenv("PEERPODS_NAMESPACE")
+
+	// Add annotations to inject ca bundle into the webhook config
+	annotations := map[string]string{
+		"service.beta.openshift.io/inject-cabundle": "true",
+	}
+
+	// Add mutating webhook configuration
+	mutatingWebhookConfig := &admissionregistrationv1.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: webhookConfigName,
+			// Add annotations
+			Annotations: annotations,
+		},
+		Webhooks: []admissionregistrationv1.MutatingWebhook{
+			{
+				Name: webhookName,
+				ClientConfig: admissionregistrationv1.WebhookClientConfig{
+					Service: &admissionregistrationv1.ServiceReference{
+						Name:      webhookSvcName,
+						Namespace: webhookSvcNamespace,
+						Path:      &webhookPath,
+					},
+				},
+				// Add rules
+				Rules: []admissionregistrationv1.RuleWithOperations{
+					{
+						Operations: []admissionregistrationv1.OperationType{
+							admissionregistrationv1.Create,
+							admissionregistrationv1.Update,
+						},
+						Rule: admissionregistrationv1.Rule{
+							APIGroups:   []string{""},
+							APIVersions: []string{"v1"},
+							Resources:   []string{"pods"},
+						},
+					},
+				},
+				// Add failure policy
+				FailurePolicy: &failurePolicy,
+				// Add side effects
+				SideEffects: &sideEffect,
+				// Add admission review versions
+				AdmissionReviewVersions: []string{"v1"},
+				// Add namespace selector using matchExpressions
+				NamespaceSelector: &metav1.LabelSelector{
+					MatchExpressions: []metav1.LabelSelectorRequirement{
+						{
+							Key:      "kubernetes.io/metadata.name",
+							Operator: metav1.LabelSelectorOpNotIn,
+							// Take values from a predefined list
+							Values: namespacesToExclude,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Create MutatingWebhookConfiguration object
+	if err := r.Client.Create(context.Background(), mutatingWebhookConfig); err != nil {
+		if !k8serrors.IsAlreadyExists(err) {
+			r.Log.Error(err, "Failed to create MutatingWebhookConfiguration object")
```

Comment:
The text is okay but the caller is also logging the error. Do we need both ? This is a general question for the rest of the patch actually.

Reply:
Also is it worth a call to `Error()` or `Info()` should be enough ?